### DISCUSSION TITLE
GH action does not respect nested/multiple negative condtion from reusable workflow

### DISCUSSION THREAD

**Discussion by suhmu** [community/community#64514] [Actions] [answered]
### Select Topic Area

Bug

### Body

Recently we  have encountered that multiple negative conditions with **if** from reusable workflows is not being respected by github action while this had been working for quite sometime. 

Example:

snippet from resusable workflow:
`if: github.event_name != 'pull_request' || ${{ matrix.os != 'windows-latest' }}`

It was all working fine but now workflow gets triggered even with `${{ matrix.os != 'windows-latest' }}`.
It works fine with below:
- single negative condition
- multiple match or `==` conditions.

while fails for below
- it fails for multiple negative or `!=`

**Comment by arafajs** [comment]
Try simplifying the conditions to use positive checks instead of negatives where possible. For example:
`if: github.event_name == 'push' && matrix.os == 'ubuntu-latest'`

- Use && instead of || for the negative checks, as De Morgan's laws state that not (A || B) is equivalent to not A && not B.
- Split the conditions into separate if statements, like:
```
if: github.event_name != 'pull_request'  
if: matrix.os != 'windows-latest'
```

**Reply by suhmu** [reply, marked as answer]
@arafajs Thanks for the suggestion. 
Yes, this is something we have followed for a fix:
`if: github.event_name == 'push' && matrix.os == 'ubuntu-latest'`
